src/librbd/api/DiffIterate.cc

```diff
@@ -456,35 +457,40 @@ int DiffIterate<I>::diff_object_map(uint64_t from_snap_id, uint64_t to_snap_id,
       return -EINVAL;
     }
     object_map.resize(num_objs);
+    object_diff_state->resize(object_map.size());
 
     uint64_t overlap = std::min(object_map.size(), prev_object_map.size());
-    for (uint64_t i = 0; i < overlap; ++i) {
+    auto it = object_map.begin();
+    auto overlap_end_it = it + overlap;
+    auto pre_it = prev_object_map.begin();
+    auto diff_it = object_diff_state->begin();
+    uint64_t i = 1;
```

Comment:
Why does this start at index 1?

Reply:
i just mistake object_id start from 1, already changed to `uint64_t i = 0;`